### PR TITLE
Fix many build warning and issues, more documentation additions [skip ci]

### PR DESCRIPTION
**Pull request by nvkevlu**
Fixes many apidocs build warnings throughout the apidocs. Fix and improve sections of the official docs.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def resolve_xref(self, env, fromdocname, builder, typ, target, node, contnode):
 html_theme_options = {
     "collapse_navigation": True,
     "display_version": True,
-    "navigation_depth": 5,
+    "navigation_depth": 8,
```
Comment (YuanTingHsieh) [inline docs/conf.py:106]: Just curious, what does this change?

File: docs/examples/federated_statistics.rst [hunk 2]
```diff
@@ -57,7 +57,7 @@ The main steps are:
     * provide server side configuration to specify target statistics and their configurations and output location
     * implement the local statistics generator (statistics_spec)
     * provide client side configuration to specify data input location
-    * The detailed example instructions can be found in `Data frame statistics <https://github.com/NVIDIA/NVFlare/tree/main/examples/federated_statistics/df_stats/README.md>`_
+    * The detailed example instructions can be found in `Data frame statistics <https://github.com/NVIDIA/NVFlare/tree/main/examples/federated_statistics/df_stats/README.md>`__
```
Comment (YuanTingHsieh) [inline docs/examples/federated_statistics.rst:60]: I thought the syntax for link is one underscore?
What does two underscore mean?

**Review by chesterxgchen — APPROVED** [review]
LGTM

File: docs/conf.py (hunk 1 above)
Comment (nvkevlu) [reply docs/conf.py:106]: This allows for the navigation to expand more, so the submodules in the apidocs will be more navigable.

File: docs/examples/federated_statistics.rst (hunk 2 above)
Comment (nvkevlu) [reply docs/examples/federated_statistics.rst:60]: Two underscores makes the reference anonymous I think, making it so if you have duplicate links, the build doesn't complain about duplicates.

**Pull request merged**